### PR TITLE
Set process limit for the omsagent user

### PR DESCRIPTION
**Pull request by lagalbra**
Maximum number of processes found during customer and in-house testing was 28; setting default to 75 to be safe
Give option in omsadmin.sh to set and reset the limit
Remove duplicate onboarding system tests

pBuild and system tests passed
Verified on a test machine by setting the limit to 10; no new processes were started. Main FluentD process continued to run.

@Microsoft/omsagent-devs

### CODE REVIEW

File: test/installer/scripts/test_omsadmin.sh [hunk 1]
```diff
@@ -1,42 +0,0 @@
-BASE_DIR=$1
-RUBY_TESTVERS=$2
```
Comment (NarineM) [inline test/installer/scripts/test_omsadmin.sh]: This test was for testing omsadmin.sh shell script with onboarding parameters. why you think it has to be deleted?

File: test/installer/scripts/omsadmin_systest.rb [hunk 2]
```diff
@@ -130,4 +131,45 @@ def test_onboard_two_workspaces
     assert_match(/No Workspace/, output_list, "No workspace should be listed: #{output_list}")
   end
 
+  def test_set_proc_limit_default
```
Comment (NarineM) [inline test/installer/scripts/omsadmin_systest.rb]: Please add a test for proc minimum.

File: installer/scripts/omsadmin.sh [hunk 3]
```diff
@@ -352,6 +367,25 @@ set_proxy_setting()
     fi
 }
 
+set_omsagent_proc_limit()
```
Comment (NarineM) [inline installer/scripts/omsadmin.sh]: I would expect this API to have a minimum accepted limit.

File: test/installer/scripts/test_omsadmin.sh (hunk 1 above)
Comment (lagalbra) [inline test/installer/scripts/test_omsadmin.sh]: [omsadmin_systest.rb](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/test/installer/scripts/omsadmin_systest.rb#L60) already covers this test case and more.

File: installer/scripts/omsadmin.sh (hunk 3 above)
Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: Why? And if I add it, what should the minimum be? If a customer is setting this limit, they already are familiar with how processes work, and know that setting the limit too low may hinder functionality.

File: test/installer/scripts/test_omsadmin.sh (hunk 1 above)
Comment (NarineM) [inline test/installer/scripts/test_omsadmin.sh]: If that is the case, please update the PR description to state "duplicate" instead of "redundant".

File: installer/scripts/omsadmin.sh (hunk 3 above)
Comment (NarineM) [inline installer/scripts/omsadmin.sh]: Customer would not know what is the minimum accepted limit where the agent can still function.

Comment (NarineM) [inline installer/scripts/omsadmin.sh]: I am going to leave this PR for someone else to review who was initially involved with ulimit design discussions. I am still not clear what would be omsagent behavior if customer puts 1 as proc limit and how that can impact omiagent?

Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: I have not tested with numbers as small as 1; I will set a minimum limit of 5 to safeguard against unwise input.

File: test/installer/scripts/test_omsadmin.sh (hunk 1 above)
Comment (lagalbra) [inline test/installer/scripts/test_omsadmin.sh]: PR description updated.

File: test/installer/scripts/omsadmin_systest.rb (hunk 2 above)
Comment (lagalbra) [inline test/installer/scripts/omsadmin_systest.rb]: Added with `test_set_proc_limit_below_min`; testing now

**Comment by lagalbra** [comment]
@keikhara Can you please review, since you were part of the design discussions?

**Comment by lagalbra** [comment]
Test for minimum ulimit added and passed. @NarineM Please review.

File: installer/scripts/omsadmin.sh [hunk 4]
```diff
@@ -346,6 +362,28 @@ set_proxy_setting()
...
```
Comment (Atchub) [inline installer/scripts/omsadmin.sh]: can we apply at group level or do we need to include nxautomation user too as it is also a user created by OMS agent

Comment (Atchub) [inline installer/scripts/omsadmin.sh]: does nproc restrict for shell commands executed with root permissions? just checking if there is a constraint

Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: The scope of this task was to control the number of processes under the omsagent user specifically.
The hybrid worker should be able to set a different limit for the nxautomation user if they need to in the future.

Comment (NarineM) [inline installer/scripts/omsadmin.sh]: Good point, when reviewing I only taken omsconfig and omsagent process related users into an account.
@lagalbra , can you look into Atchut's concern?

Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: I believe that would run as root, so this limit wouldn't restrict it, no.

Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: To answer Atchut's question a little further, yes, these limits can be placed at the group level.
To answer Narine's question, no. The owners of the the nxautomation user should place limits if desired, since they have any and all context for what the limits should be. For now, the customer is only looking for limits on the omsagent user.

Comment (NarineM) [inline installer/scripts/omsadmin.sh]: By omsagent user they most likely meant the users that are used to perform operations within or by omsagent related process , so I would think processed spinned up by nxautomation will also count.

Comment (lagalbra) [inline installer/scripts/omsadmin.sh]: Spoke with @keikhara ; our aim is to make the resource limits as specific as possible. If we place the limit at the group level instead of the user level, the omsagent could be pointlessly throttled when the nxautomation user misbehaves.

I will send out an email asking the automation team to add a process limit for the nxautomation worker, after they have performed similar testing for their user's processes.

Given this, please approve this PR.

**Comment by lagalbra** [comment]
We do not have any system tests against the agent service; I have verified this end-to-end myself.

The agent service does not stop if you set the limit to a small number. It's already running, so it keeps running, but new processes over that number cannot be started.

**Comment by NarineM** [comment]
we should be adding new tests for this feature work.I do not think unit test level coverage is enough.

**Comment by lagalbra** [comment]
Rebased against master. Will merge once Travis completes.

**Comment by lagalbra** [comment]
I've verified on a test machine that the agent can be successfully restarted when the ulimit is set to 25.
